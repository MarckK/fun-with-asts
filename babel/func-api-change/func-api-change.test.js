const plugin = require('./func-api-change')
const pluginTester = require('babel-plugin-tester')

pluginTester({
  plugin,
  tests: {
    'does not change non importantFunc invocations': 'someFunc(a, b, c);',
    'changes importantFunc invocations from ordinal to object - params as references': {
      code: 'importantFunc(a,b,c)',
      snapshot: true
    },
    'changes importantFunc invocations from ordinal to object - params as literals': {
      code: 'importantFunc("mike", 32, "mpls")',
      snapshot: true
    }
  }
})