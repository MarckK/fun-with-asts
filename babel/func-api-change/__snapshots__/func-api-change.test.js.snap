// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2. changes importantFunc invocations from ordinal to object - params as references 1`] = `
"
importantFunc(a,b,c)

      ↓ ↓ ↓ ↓ ↓ ↓

importantFunc({
  name: a,
  age: b,
  location: c
});
"
`;

exports[`3. changes importantFunc invocations from ordinal to object - params as literals 1`] = `
"
importantFunc(\\"mike\\", 32, \\"mpls\\")

      ↓ ↓ ↓ ↓ ↓ ↓

importantFunc({
  name: \\"mike\\",
  age: 32,
  location: \\"mpls\\"
});
"
`;
